### DISCUSSION TITLE
Log only if causer_type is not null

### DISCUSSION THREAD

**Discussion by incoming-th** [spatie/laravel-activitylog#857] [Q&A] [answered]
Hi,

Trying to figure out how it is possible to filter the logs to only save those where the causer_type/causer_id is not null.

I have lot of jobs modifying the Models and I don't need to have those saved in the DB.
![Screenshot 2021-03-04 154204](https://user-images.githubusercontent.com/26375917/109935622-5c7b7980-7d00-11eb-8287-d4ba84bdad77.jpg)
Hope anyone will guide me in the right direction.

Thanks

**Comment by Gummibeer** [comment, marked as answer]
You can disable logging for a whole runtime or single callbacks/code snippets.
https://spatie.be/docs/laravel-activitylog/v3/advanced-usage/disabling-logging

#661 would also be a solution to your problem.

**Comment by boryn** [comment]
Hi @MickaelTH !

I have come up with this solution, specifying an early "empty" return at `getActivitylogOptions()`:

```php
if (Auth::user() === null) {
    return LogOptions::defaults()
                     ->dontSubmitEmptyLogs();
}
```
